build-tools/packages/build-cli/src/commands/setReleaseTagForPublishing.ts

```diff
@@ -0,0 +1,104 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Flags } from "@oclif/core";
+import { Package, updatePackageJsonFile } from "@fluidframework/build-tools";
+import path from "node:path";
+import { PackageCommand } from "../BasePackageCommand";
+import * as fs from "fs";
+
+/**
+ * A type defining the four release types
+ */
+type UpdatePackageJsonType = "alpha" | "beta" | "public";
+
+/**
+ * A typeguard to check if a string is a {@link UpdatePackageJsonEnum}.
+ */
+function isUpdatePackageJsonTypes(type: string): type is UpdatePackageJsonType {
+	return type === "alpha" || type === "beta" || type === "public";
+}
+
+export default class SetReleaseTagPublishingCommand extends PackageCommand<
+	typeof SetReleaseTagPublishingCommand
+> {
+	static readonly description =
+		"Updates the types in package.json based on the release flag. This command is intended for use in publishing pipelines and may not be intended for regular developer use.";
+
+	static flags = {
+		types: Flags.custom<UpdatePackageJsonType>({
+			description:
+				"The types flag is used to specify the type of release when updating a package.json file. It accepts a custom enumeration called UpdatePackageJsonEnum that defines different release types.",
+			parse: async (input) => {
+				if (isUpdatePackageJsonTypes(input)) {
+					return input;
+				}
+				throw new Error(`Invalid release type: ${input}`);
+			},
+		})(),
+		...PackageCommand.flags,
+	};
+
+	private readonly packageTypesList: string[] = [];
+
+	protected async processPackage(pkg: Package): Promise<void> {
+		// Define paths to api-extractor.json
+		const apiExtractorConfigFilePath = path.join(pkg.directory, "api-extractor.json");
+
+		// Check if api-extractor.json exists in the package directory
+		const apiExtractorConfigExists = fs.existsSync(apiExtractorConfigFilePath);
+
+		updatePackageJsonFile(pkg.directory, (json) => {
+			const types: string | undefined = json.types ?? json.typings;
+
+			if (types === undefined) {
+				return;
+			}
+
+			const [typesFolder] = types.split("/");
+
+			const includesAlphaBeta: boolean = types.includes("alpha") || types.includes("beta");
+
+			if (apiExtractorConfigExists) {
+				// Read the content of api-extractor.json
+				const apiExtractorConfig: string = fs.readFileSync(
+					apiExtractorConfigFilePath,
+					"utf-8",
+				);
```

Comment:
> subsequently examines its rollup configurations

It also should handle the inherited settings case, where a config inherits the rollup config from another config file.

Reply:
`ExtractorConfig.laodFile(path)` loads the api-extractor.json config file from the specified file path. If the `extends` field is present, the referenced file will be merged. `extends` keywords exists in most of the api-extractor.json files:
```ts
{
	"$schema": "https://developer.microsoft.com/json-schemas/api-extractor/v7/api-extractor.schema.json",
	"extends": "@fluidframework/build-common/api-extractor-common-report.json"
}
```
The problem is it cannot find the module `@fluidframework/build-common/api-extractor-common-report.json`. I'm not sure as to how this can be resolved